Week05/lesson10/homework/cash_register_starter_code/js/index.js

```diff
@@ -0,0 +1,8 @@
+function add (){
+  var newEntry = parseFloat(document.getElementById("#newEntry").value);
+  var total = "newEntry" + "total";
+
+  document.querySelector("total").innerHTML = "Result";
+}
+
+return = result;
```

Comment:
I think maybe your overarching "problem" (and I hesitate massively to use that word because you're doing REALLY well!!) is that you rush into things a little bit. 

You need to treat the HWs the way I do the assignments in class, I always check if things are hooked up first via `console.log` and then and only then do I proceed. Some pseudo code would also really help you I think.

So if you have time, take a step back and try with this assignment again. I'll give you a jumping-off point:

Step 1) Make sure `index.js` is properly connected to `index.html` by using a `console.log("hello world");` at the very top of the JS file (the rest of the JS file should be empty). That works, cool, now:

Step 2)  Get the data that connects to a moment in time using (`document.querySelector`):
```
document.querySelector("#entry")
```

Step 3) What is the moment in time? `#entry` is a form. Forms get submitted, so hook that up to some sort of action:

```
function doSomething() {
  console.log("doing something");
}

document.querySelector("#entry").onsubmit = doSomething;
```

Cool, now I'm seeing `doing something` in the console really briefly after I hit enter (thus submitting the form).

Now, Nico mentioned that if forms aren't given a "Thank you" page to redirect to, their default behaviour is to refresh. That's probably why I'm seeing the `doing something` in the console for such a short amount of time. She explained some mumbo jumbo about an `event` and how you have to prevent its default behaviour, let me just copy her code in:

```
function doSomething(event) {
  event.preventDefault();
  console.log("doing something");
}

document.querySelector("#entry").onsubmit = doSomething;
```

Cool now I'm seeing `doing something` when I hit enter without it randomly disappearing really fast.

Step 4) If needed, grab more data. Then modify said data as required.

What data will I need? The value of the input (whose id is `newEntry`). How will I modify it? Well, I need to use it to keep a list of items as well as a running total.

And I think I'll leave you here Jo, good luck! You can do this!